controllers/cloud.redhat.com/clowdapp_controller.go

```diff
@@ -77,9 +77,10 @@ const appFinalizer = "finalizer.app.cloud.redhat.com"
 // ClowdAppReconciler reconciles a ClowdApp object
 type ClowdAppReconciler struct {
 	client.Client
-	Log      logr.Logger
-	Scheme   *runtime.Scheme
-	Recorder record.EventRecorder
+	Log                logr.Logger
+	Scheme             *runtime.Scheme
+	Recorder           record.EventRecorder
+	ReconcileStartTime time.Time
```

Comment:
My only concern here is that this struct becomes large with a bunch of new fields, is there a category we could put it under, almost everything else is really core/crucial,k any ideas?

Reply:
I could factor reconciliation metrics out into its own struct and use that in `ClowdAppReconciler.` It could track its own state. That would be the cleaner design I think.

Reply:
Factored it out, see new commits after checks pass